### PR TITLE
use typeorm migrations for quickstart

### LINKED ISSUE #6: TypeORM migrations for quickstart

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+This [template repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template) follows this IaSQL [tutorial](https://docs.iasql.com/quickstart/) to deploy a Node.js HTTP server within a docker container on your AWS account using ECS, ECR and ELB. The container image will be hosted as a public repository in ECR and deployed to ECS using Fargate.
+
+
+## Migrations
+
+To add a migration to infra as a SQL db run:
+
+```bash
+npm i
+./node_modules/.bin/typeorm migration:create --outputJs -n <YourMigrationName>
```
Comment (aguillenv) [inline README.md]: if I'm not wrong you can do
```suggestion
npm i
npx typeorm migration:create --outputJs -n <YourMigrationName>
```

**Review by dfellis — APPROVED** [review]

File: infra/src/migration/1646683871219-Initial.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      INSERT INTO aws_task_definition ("family", cpu_memory) VALUES ('${TASK_DEF_FAMILY}', '${TASK_DEF_RESOURCES}');
+
+      INSERT INTO aws_container_definition ("name", public_repository_id, task_definition_id, log_group_id, tag, essential, memory_reservation, host_port, container_port, protocol)
+      VALUES (
+        '${CONTAINER}',
+        (select id from aws_public_repository where repository_name = '${REPOSITORY}' limit 1),
+        (select id from aws_task_definition where family = '${TASK_DEF_REPO_FAMILY}' and status is null limit 1),
+        (select id from log_group where log_group_name = '${LOG_GROUP}' limit 1),
+        '${IMAGE_TAG}', ${CONTAINER_ESSENTIAL}, ${CONTAINER_MEM_RESERVATION}, ${HOST_PORT}, ${CONTAINER_PORT}, '${PROTOCOL.toLowerCase()}'
+      );
```
Comment (aguillenv) [inline infra/src/migration/1646683871219-Initial.js]: Same

File: infra/src/migration/1646683871219-Initial.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+      INSERT INTO aws_security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+      SELECT false, 'tcp', ${PORT}, ${PORT}, '0.0.0.0/0', '${SECURITY_GROUP}', id
+      FROM aws_security_group
+      WHERE group_name = '${SECURITY_GROUP}';
+
+      INSERT INTO aws_security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+      SELECT true, 'tcp', 443, 443, '0.0.0.0/0', '${SECURITY_GROUP}', id
+      FROM aws_security_group
+      WHERE group_name = '${SECURITY_GROUP}';
```
Comment (aguillenv) [inline infra/src/migration/1646683871219-Initial.js]: Ideally this should be in a transaction (BEGIN; ... COMMIT;)

File: infra/src/migration/1646683871219-Initial.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    // TODO replace SPs
+    await queryRunner.query(`
+      INSERT INTO aws_target_group (target_group_name, target_type, protocol, port, vpc, health_check_path)
+      VALUES ('${TARGET_GROUP}', 'ip', 'HTTP', ${PORT}, '${DEFAULT_VPC}', '${TARGET_GROUP_HEALTH_PATH}');
+      call create_or_update_aws_load_balancer(
+        '${LOAD_BALANCER}', 'internet-facing', '${DEFAULT_VPC}', 'application', 'ipv4', array['${SECURITY_GROUP}']
+      );
+      call create_or_update_aws_listener('${LOAD_BALANCER}', ${PORT}, 'HTTP', 'forward', '${TARGET_GROUP}');
+      INSERT INTO log_group (log_group_name)
+      VALUES ('${LOG_GROUP}');
```
Comment (aguillenv) [inline infra/src/migration/1646683871219-Initial.js]: Same

File: infra/src/migration/1646683871219-Initial.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      CALL delete_aws_target_group('${TARGET_GROUP}');
+      DELETE FROM log_group WHERE log_group_name = '${LOG_GROUP}';
+    `);
+
+    // delete security groups
+    await queryRunner.query(`
+      DELETE FROM aws_security_group_rule
+      USING aws_security_group
+      WHERE aws_security_group.id = aws_security_group_rule.security_group_id AND aws_security_group.group_name = '${SECURITY_GROUP}';
+      DELETE FROM aws_security_group WHERE group_name = '${SECURITY_GROUP}';
```
Comment (aguillenv) [inline infra/src/migration/1646683871219-Initial.js]: Same

**Review by aguillenv — APPROVED** [review]

File: infra/src/migration/1646683871219-Initial.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        INSERT INTO aws_cluster (cluster_name) VALUES('${CLUSTER}');
+
+        INSERT INTO aws_task_definition ("family", cpu_memory)
+        VALUES ('${TASK_DEF_FAMILY}', '${TASK_DEF_RESOURCES}');
+
+        INSERT INTO aws_container_definition ("name", public_repository_id, task_definition_id, log_group_id, tag, essential, memory_reservation, host_port, container_port, protocol)
+        VALUES (
+          '${CONTAINER}',
+          (select id from aws_public_repository where repository_name = '${REPOSITORY}' limit 1),
+          (select id from aws_task_definition where family = '${TASK_DEF_REPO_FAMILY}' and status is null limit 1),
```
Comment (aguillenv) [inline infra/src/migration/1646683871219-Initial.js]: Here you are using 2 different task definitions families. You need to stick to one of them.

File: infra/src/migration/1646683871219-Initial.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    // delete ECS + ECR
+    await queryRunner.query(`    
+      BEGIN;
+        DELETE FROM aws_container_definition
+        USING aws_task_definition
+        WHERE aws_container_definition.task_definition_id = aws_task_definition.id and aws_task_definition.family = '${TASK_DEF_REPO_FAMILY}';
+
+        DELETE FROM aws_container_definition
+        USING aws_public_repository
+        WHERE aws_container_definition.public_repository_id = aws_public_repository.id and aws_public_repository.repository_name = '${REPOSITORY}';
```
Comment (aguillenv) [inline infra/src/migration/1646683871219-Initial.js]: This is not necessary I think?

File: infra/src/migration/1646683871219-Initial.js (hunk 6 above)
Comment (aguillenv) [reply infra/src/migration/1646683871219-Initial.js]: I'd replace `TASK_DEF_REPO_FAMILY` with `TASK_DEF_FAMILY` all over the migration.

Comment (depombo) [reply infra/src/migration/1646683871219-Initial.js]: that was it part of the issue, thank you!

File: infra/src/migration/1646683871219-Initial.js (hunk 7 above)
Comment (depombo) [reply infra/src/migration/1646683871219-Initial.js]: No, removed!

**Pull request merged**